src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -370,7 +373,33 @@ public static void checkApiUrlValidity(@Nonnull GitHub gitHub, @CheckForNull Sta
             gb.withEndpoint(apiUrl);
             gb.withRateLimitHandler(CUSTOMIZED);
 
+
+            Interceptor interceptor = new Interceptor() {
+                @Override
+                public Response intercept(Chain chain) throws IOException {
+                    Request request = chain.request();
+
+                    // okhttp caches the tag response even if the tag object is not found.
+                    // tags are mutable, therefore we should not cache the references
```

Comment:
Wait... no.  
Most information provided by the github api is mutable.  
We can't just turn off caching for all tag queries.  If caching isn't working on tags but is working for other items, this is probably an issue with the `Cache-Control` or `ETag` values provided by the the server or a bug in the Caching implemented in okhttp2, not something that Jenkins is doing wrong.  

There needs to be a lot more data gathered about what is causing this before we take this step.